### PR TITLE
Refactor ES Client methods

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
## Part of https://github.com/elastic/enterprise-search-team/issues/6412

Current structure of our classes that call Elasticsearch is neat, but there are some minor problems here and there as I see them. This PR attempts to refactor out these problems.

It contains several connected changes, might make it a bit harder to review, but they all follow the same logic.

Changes are:

### Change 1: Instead of having one ESClient, there's now additionally ESManagementClient.

A lot of classes inherit from ESClient, for example ConnectorIndex class. ConnectorIndex class is used to do certain operations on connectors - get all connectors, read one connector, get supported connectors, etc.

ESClient, however, had methods that are touching things that logically should not be available for _specialized_ classes that inherit from it, for example this method:

```
async def delete_indices(self, indices):
    # logic here
```

It seems illogical that `ConnectorIndex` class that's a rich wrapper for operations with connectors index can have access to deleting arbitrary indices.

Based on that logic, I've moved all methods that allow for arbitrary index management to `ESManagementClient` class - now it's responsible for management operations - create indices if they don't exit, clean their content, delete indices, etc.

### Change 2: Move ESClient.get_connector_doc to ConnectorIndex.get_connector_by_index

It's a logical change as this is a connector access operation, it does not make sense to have it in `ESClient` class that's a base class for lots of other non-connector-specific classes

### Change 3: ensure_exists method does not create/delete documents now, just uses built-in client method

Self-explanatory - instead of creating indices with `PUT {_index}/_doc` and then deleting this doc, we can just call `client.indices.create` method.

This change could have been separated, but I squeezed it here, sorry.

All other changes are pretty much code/test changes to accomodate the above 3 changes.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/services/job_cleanup.py [hunk 1]
```diff
@@ -27,6 +28,7 @@ def __init__(self, config):
     async def _run(self):
         logger.debug("Successfully started Job cleanup task...")
         self.connector_index = ConnectorIndex(self.es_config)
+        self.es_management_client = ESManagementClient(self.es_config)
```
Comment (vidok) [inline connectors/services/job_cleanup.py:31]: We use different variable (`es_client`) names in other places. It's a nit request but I would make it the same for all the classes.

**Review by vidok — APPROVED** [review]
LGTM

File: connectors/services/job_cleanup.py (hunk 1 above)
Comment (artem-shelkovnikov) [reply connectors/services/job_cleanup.py:31]: So use `es_client` for `ESManagementClient` too, or have `es_management_client` variable?

Comment (vidok) [reply connectors/services/job_cleanup.py:31]: I don't have a strong opinion on that :). I think `es_management_client`  is a little more explicit so I would use it but it's up to you. I'm fine with `es_client` too.

Comment (artem-shelkovnikov) [reply connectors/services/job_cleanup.py:31]: Makes sense! I've addressed this in https://github.com/elastic/connectors/pull/1999/commits/bf1ef636637bbe307d2e81d8b0b1237db02b7a4c

**Review by vidok — APPROVED** [review]
Great work!

**Pull request merged**